BDArmory/Ammo/PooledBullet.cs

```diff
@@ -1495,6 +1495,246 @@ IEnumerator DelayedDetonationRoutine()
+                pBullet.bulletDrop = drop;
+
+                if (bulletType.tntMass > 0)
+                {
+                    pBullet.HEType = eHEType;
+                }
+                else
+                {
+                    pBullet.HEType = PooledBulletTypes.Slug;
+                }
+
+                if (bulletType.tntMass > 0 || (!isSubP && bulletType.beehive))
+                {
+                    pBullet.tntMass = bulletType.tntMass;
+                    pBullet.explModelPath = graphicsInfo.explModelPath;
+                    pBullet.explSoundPath = graphicsInfo.explSoundPath;
+                    pBullet.detonationRange = detRange;
+                    pBullet.timeToDetonation = detTime;
+                    pBullet.fuzeType = efuzeType;
+                }
+                else
+                {
+                    pBullet.fuzeType = BulletFuzeTypes.None;
+                    pBullet.sabot = sabotBool;
+                }
+
+                pBullet.EMP = bulletType.EMP;
+                pBullet.nuclear = bulletType.nuclear;
+                if (pBullet.nuclear) // Inherit the parent shell's nuke models.
+                {
+                    pBullet.flashModelPath = nukeInfo.flashModelPath;
+                    pBullet.shockModelPath = nukeInfo.shockModelPath;
+                    pBullet.blastModelPath = nukeInfo.blastModelPath;
+                    pBullet.plumeModelPath = nukeInfo.plumeModelPath;
+                    pBullet.debrisModelPath = nukeInfo.debrisModelPath;
+                    pBullet.blastSoundPath = nukeInfo.blastSoundPath;
+                }
+                
+                // No sub-sub projectiles!
+                pBullet.beehive = bulletType.beehive && !isSubP;
+                if (bulletType.beehive && !isSubP)
+                {
+                    pBullet.subMunitionType = bulletType.subMunitionType;
+                }
+
+                pBullet.impulse = bulletType.impulse;
+                pBullet.massMod = bulletType.massMod;
+
+                //pBullet.homing = BulletInfo.homing;
+                pBullet.dragType = eDragType;
+                
+                pBullet.tgtShell = targetShell;
+                pBullet.tgtRocket = targetRocket;
+
+                pBullet.bullet = bulletType;
+                pBullet.stealResources = steal;
+                pBullet.dmgMult = damageMult;
+                pBullet.targetVessel = tgtVessel;
+                pBullet.guidanceDPS = guidance;
+                pBullet.isSubProjectile = isSubP;
+                pBullet.isAPSprojectile = isAPSP;
+                pBullet.gameObject.SetActive(true);
+                
+                if (isSubP)
+                    pBullet.SetTracerPosition();
+
+                if (registerShot)
+                    BDACompetitionMode.Instance.Scores.RegisterShot(sourceInfo.vessel.GetName());
+
+                if (!pBullet.CheckBulletCollisions(timestep)) // Check that the bullet won't immediately hit anything and die.
```

Comment:
Thanks for the breakdown on Krakensbane. Regarding the condition, I think a `!addSourcePartVel` will probably suffice for this and future cases where similar issues might pop up, since the only time the latter branch would need to be considered is, like you said, when bullets are launched from a weapon and require adding the source part velocity. I can't think of any situation where we'd need to have more complex logic involving both `isSubP` and `addSourcePartVel`.